### PR TITLE
Ios/booking

### PR DESCRIPTION
**Pull request by Daniel-Nielsen**

### CODE REVIEW

**Comment by Daniel-Nielsen** [comment]
@marksdk other than moving it into the beta doc site, have you had a chance to look at the content?

File: collections/_ios/v3/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,12 @@ Changelog for MapsIndoors for iOS. This document structure is based on [Keep a C
 ### Removed
 -->
 
+## [3.15.0] 2020-11-12
+
+### Added
+
+- Support for Booking of Locations through a MapsIndoors Google Calendar Booking Provider, see [guide about booking](https://mapsindoors.github.io/ios/v3/booking/).
```
Comment (marksdk) [inline collections/_ios/v3/CHANGELOG.md:23]:
```suggestion
- Support for Booking of Locations through a MapsIndoors Google Calendar Booking Provider, see [guide about Booking](https://mapsindoors.github.io/ios/v3/booking/).
```

File: collections/_ios/v3/booking/booking-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+guide_title: booking-tutorial
+nav_weight: 340
+published: true
+date: 2020-11-25
+permalink: /ios/v3/booking-tutorial/
+---
+
+In this tutorial we will create a Booking experience using the Booking Service in MapsIndoors. You will learn how to list bookable Locations, list Bookings for a Location and perform new Bookings using the Booking Service, `MPBookingService`.
+
+Please note that a MapsIndoors dataset can only have bookable resources if an integration with a booking provider exists. Current examples of booking providers are _Google Calendar_ and _Microsoft Office 365_. These providers and more can be added and integrated to your MapsIndoors project by request. It is a prerequisite for this tutorial that the API key used refers to a dataset containing bookable Locations.
```
Comment (marksdk) [inline collections/_ios/v3/booking/booking-tutorial.md:13]:
```suggestion
Please note that a MapsIndoors dataset can only have bookable resources if an integration with a booking provider exists. Current examples of booking providers are _Google Calendar_ and _Microsoft Office 365_. These providers, and more, can be added and integrated to your MapsIndoors project by request. It is a prerequisite for this tutorial that the API key used refers to a dataset containing bookable Locations.
```

File: collections/_ios/v3/booking/booking-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+```swift
+override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    let cell = UITableViewCell.init()
+    let location = bookableLocations[indexPath.row]
+    cell.textLabel?.text = location.name
+    return cell
+}
+```
+
+In `tableView(:didSelectRowAt indexPath:)`, get the relevant `MPLocation` for the `indexPath`. Initialise a `BookingsController` which we will implement next. Assign the selected location to a `bookableLocation` property on `BookingsController` and push the controller to the navigation stack.
```
Comment (marksdk) [inline collections/_ios/v3/booking/booking-tutorial.md:78]:
```suggestion
In `tableView(:didSelectRowAt indexPath:)`, get the relevant `MPLocation` for the `indexPath`. Initialise a `BookingsController` which we will implement next. Assign the selected Location to a `bookableLocation` property on `BookingsController` and push the controller to the navigation stack.
```

File: collections/_ios/v3/booking/index.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+bookableQuery.startTime = Date.init()
+bookableQuery.endTime = bookableQuery.startTime.advanced(by: 60*60)
+
+bookingService.getBookableLocations(using: bookableQuery, completion: { (locations, error) in
+
+})
+```
+
+The above example creates a query for locations that are bookable for a timespan between now and 1 hour ahead.
```
Comment (marksdk) [inline collections/_ios/v3/booking/index.md:44]:
```suggestion
The above example creates a query for Locations that are bookable for a timespan between now and 1 hour ahead.
```

File: collections/_ios/v3/booking/index.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+bookingsQuery.location = myRoomLocation
+bookingsQuery.startTime = Date.init().advanced(by: -60*60)
+bookingsQuery.endTime = bookingsQuery.startTime?.advanced(by: 24*60*60)
+
+bookingService.getBookingsUsing(bookingsQuery) { (bookings, error) in
+}
+```
+
+The above example creates a query for bookings that exists for a location with timespan between 1 hour ago and 24 hours ahead.
```
Comment (marksdk) [inline collections/_ios/v3/booking/index.md:70]:
```suggestion
The above example creates a query for bookings that exists for a Location with timespan between 1 hour ago and 24 hours ahead.
```

**Review by marksdk — DISMISSED** [review]

**Comment by marksdk** [comment]
@Daniel-Nielsen *Great* job on this! I'm stealing it for the new docs-site, so hold off on merging just yet 🙂

**Pull request closed without merging**